src/main/java/com/synopsys/integration/detect/workflow/componentlocationanalysis/ScanResultToComponentListTransformer.java

```diff
@@ -1,16 +1,50 @@
 package com.synopsys.integration.detect.workflow.componentlocationanalysis;
 
 import com.synopsys.integration.blackduck.api.generated.view.DeveloperScansScanView;
-import java.util.List;
+import com.synopsys.integration.fixpr.generic.beans.domain_objects.Component;
+
+import java.util.*;
 
 /**
  * Transforms a list of {@link DeveloperScansScanView} to a list of {@link Component}s, which will then be used to
  * assemble the input to Component Locator.
  *
- * Will be fully implemented in a subsequent pull request to the Fix PR feature branch.
  */
 public class ScanResultToComponentListTransformer {
-    public ComponentLocatorInput transformScanResultToComponentList(List<DeveloperScansScanView> rapidScanFullResults) {
-        return new ComponentLocatorInput(null, null, null);
+    /**
+     * Given a list of reported components from a Rapid/Stateless Detector scan, transforms each element to its
+     * corresponding {@link Component} with appropriate metadata.
+     * @param rapidScanFullResults
+     * @return list of {@link Component}s
+     */
+    public List<Component> transformScanResultToComponentList(List<DeveloperScansScanView> rapidScanFullResults) {
+        HashMap<String, ScanMetadata> componentIdWithMetadata = new HashMap<>(); // TODO investigate if duplicates may make sense here?
+
+        for (DeveloperScansScanView component : rapidScanFullResults) {
+            componentIdWithMetadata.put(component.getExternalId(), populateMetadata(component));
+        }
+
+        return externalIDsToComponentList(componentIdWithMetadata);
+    }
+
+    private List<Component> externalIDsToComponentList(HashMap<String, ScanMetadata> componentIdWithMetadata) {
+        List<Component> componentList = new ArrayList<>();
+        for (String gav : componentIdWithMetadata.keySet()) {
+            // TODO get separator based on forge for diff pkg mngrs instead of hardcoding ":" here?
+            String[] parts = gav.split(":");
+            componentList.add(new Component(parts[0], parts[1], parts[2], componentIdWithMetadata.get(gav)));
+        }
+        return componentList;
+    }
+
+    private ScanMetadata populateMetadata(DeveloperScansScanView component) {
+        ScanMetadata remediationGuidance = new ScanMetadata();
+        // TODO add parts of "allVulnerabilities" section once blackduck-common-api version is bumped up (API v6 needed)
```

Comment:
This was requested by the Bridge team, but the Blackduck-common-api version we have does not have this piece available through DeveloperScansScanView. Will either fetch from json response or update Blackduck-common-api library -- leaving this TODO comment for now so this PR's review is not blocked.

Reply:
See IDETECT-3950